test/e2e/density.go

```diff
@@ -231,9 +235,38 @@ var _ = Describe("Density", func() {
 		}
 		itArg := testArg
 		It(name, func() {
+			uuid := string(util.NewUUID())
 			totalPods := itArg.podsPerMinion * minionCount
-			RCName = "my-hostname-density" + strconv.Itoa(totalPods) + "-" + string(util.NewUUID())
+			nameStr := strconv.Itoa(totalPods) + "-" + uuid
+			ns = "e2e-density" + nameStr
+			RCName = "my-hostname-density" + nameStr
 			RunRC(c, RCName, ns, "gcr.io/google_containers/pause:go", totalPods)
+			By("waiting for all events to be recorded")
+			last := -1
+			current := 0
+			var events *api.EventList
+			for last < current {
+				e, err := c.Events(ns).List(
+					labels.Everything(),
+					fields.Set{
+						"involvedObject.namespace": ns,
+					}.AsSelector(),
+				)
+				expectNoError(err)
+				last = current
```

Comment:
In my testing the only problems came about when a pod was continually restarted as that causes a never ending spew of events.  Even cases where there was an error the test went okay.  I dislike that notion of the test hanging because of a pod continually restarting, but atm I don't see a way to avoid it.  Either the test provides a false positive by shutting off the event collection too soon and potentially missing an issue or it hangs.  The later seemed better because it would force an investigation, but neither are desirable imo.

I could put in an insanely large timeout, like 20 minutes or something.